ecc/bls12-377/fr/iop/polynomial.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2020 ConsenSys Software Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by consensys/gnark-crypto DO NOT EDIT
+
+package iop
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/consensys/gnark-crypto/ecc/bls12-377/fr"
+)
+
+//-----------------------------------------------------
+// univariate polynomials
+
+// Enum to tell in which basis a polynomial is represented.
+type Basis int64
```

Comment:
This enums may find a better home in a dedicated `polynomial` package? 
If we consider a package importing `fr/iop`, they will read as `iop.Canonical` `iop.Regular` `iop.BitReverse` `iop.Locked`, ... which is not super clear.

From what I've seen in the PR so far, I'ld suggest (but maybe I'm wrong) to do a more "private enum" like so:
```
const (  // iota is reset to 0
        canonical = iota  // canonical == 0
        lagrange = iota  // lagrange == 1
        lagrangeCoset = iota  // lagrangeCoset == 2
)
```
And have methods defined on the `Polynomial` (if multiple `Polynomial` needs it make it an interface to implement) like `IsCanonical() bool` etc. Will be inlined by the compiler, and it makes future changes easier to refactor without user impact.

Reply:
I agree on the polynomial package, it's just that currently on Arya's branch for gkr an existing polynomial package is already used, but the plan is to eventually replace it. I'll talk with Arya about that